cli/cfncluster/cfncluster.py

```diff
@@ -206,189 +210,127 @@ def stop(args):
-            if e.message.endswith("does not exist"):
-                logger.critical(e.message)
-                sys.stdout.flush()
-                sys.exit(0)
-            else:
-                raise e
-        temp_resources.extend(resources)
-        if not resources.next_token:
-            break
-        resources = cfnconn.describe_stack_resources(stack, next_token=resources.next_token)
+    try:
+        resources = cfn.describe_stack_resource(StackName=stack_name, LogicalResourceId='MasterServer')
+        return resources.get('StackResourceDetail').get('PhysicalResourceId')
+    except ClientError as e:
+        logger.critical(e.response.get('Error').get('Message'))
+        sys.exit(1)
 
-    return [r.physical_resource_id for r in resources if r.logical_resource_id == 'MasterServer'][0]
 
 def poll_master_server_state(stack_name, config):
-    ec2conn = boto.ec2.connect_to_region(config.region,aws_access_key_id=config.aws_access_key_id,
-                                                    aws_secret_access_key=config.aws_secret_access_key)
+    ec2 = boto3.client('ec2', region_name=config.region,
+                       aws_access_key_id=config.aws_access_key_id,
+                       aws_secret_access_key=config.aws_secret_access_key)
 
     master_id = get_master_server_id(stack_name, config)
 
     try:
-        instance = ec2conn.get_only_instances(instance_ids=[master_id])[0]
-        state = instance.state
+        instance = ec2.describe_instance_status(InstanceIds=[master_id]).get('InstanceStatuses')[0]
+        state = instance.get('InstanceState').get('Name')
         sys.stdout.write('\rMasterServer: %s' % state.upper())
         sys.stdout.flush()
-        while (state != 'running' and state != 'stopped' and state != 'terminated' and state != 'shutting-down'):
+        while state not in ['running', 'stopped', 'terminated', 'shutting-down']:
             time.sleep(5)
-            state = instance.update()
+            state = ec2.describe_instance_status(InstanceIds=[master_id]).get('InstanceStatuses')[0].get('InstanceState').get('Name')
             status = ('\r\033[KMasterServer: %s' % state.upper())
             sys.stdout.write(status)
             sys.stdout.flush()
-        if (state == 'terminated' or state == 'shutting-down'):
+        if state in ['terminated', 'shutting-down']:
             logger.info("State: %s is irrecoverable. Cluster needs to be re-created.")
             sys.exit(1)
         status = ('\rMasterServer: %s\n' % state.upper())
         sys.stdout.write(status)
         sys.stdout.flush()
-    except boto.exception.BotoServerError as e:
-        if e.message.endswith("does not exist"):
-            logger.critical(e.message)
-            sys.stdout.flush()
-            sys.exit(0)
-        else:
-            raise e
+    except ClientError as e:
+        logger.critical(e.response.get('Error').get('Message'))
+        sys.stdout.flush()
+        sys.exit(1)
```

Comment:
In case the error message ends with "does not exist" we are changing the exit code from 0 to 1